pkgs/development/python-modules/xdis/default.nix

```diff
@@ -1,28 +1,35 @@
-{ stdenv
-, buildPythonPackage
-, fetchPypi
+{ lib, buildPythonPackage, fetchPypi, fetchpatch
+, click
 , pytest
 , six
-, click
 }:
 
 buildPythonPackage rec {
   pname = "xdis";
-  version = "4.0.1";
+  version = "4.0.3";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "1ifakxxawyxw4w4p58m4xdc0c955miqyaq3dfbl386ipw0f50kyz";
+    sha256 = "09fsvvh2alnbs3gv9pkx9ch4x4p0i156518hn9kg908gizgzblk0";
   };
 
+  patches = [
+    # already in master, remove when bumping from 4.0.3
+    ( fetchpatch {
+        name = "${pname}-fix-instruction-name.patch";
+        url = "https://github.com/rocky/python-xdis/commit/fa8be863f2782b8f845e7a892f8bf1d875296793.patch";
+        sha256 = "1vdxvwgw9w5rj6n6i6dyqancx7q7rm21k1ys7rq6glpyc9r6abc8";
+    })
+  ];
+
   checkInputs = [ pytest ];
   propagatedBuildInputs = [ six click ];
 
   checkPhase = ''
-    make check
+    pytest
```

Comment:
Why did you change this ?

According to https://github.com/rocky/python-xdis/blob/master/README.rst#testing `make check` seems to be the recommended way to execute the tests. I think we should prefer to do what is advised by upstream, unless there is a rational not too do so.

Reply:
because the pypi package no longer ships the `pytest` directory. See https://github.com/rocky/python-xdis/issues/51

Reply:
In such situations, I think the preferred usage is to fetch the source directly from github, which includes tests:

```patch
diff --git a/pkgs/development/python-modules/xdis/default.nix b/pkgs/development/python-modules/xdis/default.nix
index ea6f01d1a6d..4657940cbf2 100644
--- a/pkgs/development/python-modules/xdis/default.nix
+++ b/pkgs/development/python-modules/xdis/default.nix
@@ -1,4 +1,4 @@
-{ lib, buildPythonPackage, fetchPypi, fetchpatch
+{ lib, buildPythonPackage, fetchFromGitHub, fetchpatch
 , click
 , pytest
 , six
@@ -8,9 +8,11 @@ buildPythonPackage rec {
   pname = "xdis";
   version = "4.0.3";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "09fsvvh2alnbs3gv9pkx9ch4x4p0i156518hn9kg908gizgzblk0";
+  src = fetchFromGitHub {
+    owner = "rocky";
+    repo = "python-xdis";
+    rev = "${version}";
+    sha256 = "1h4j8hincf49zyd0rvn4bh0ypj8836y8vz3d496ycb9gjzkr6044";
   };
 
   patches = [
@@ -26,7 +28,7 @@ buildPythonPackage rec {
   propagatedBuildInputs = [ six click ];
 
   checkPhase = ''
-    pytest
+    make check
   '';
 
   meta = with lib; {
```

Reply:
both of them have tests, the pytest directory (from my understanding) was to just help with setup.

Reply:
i couldn't make a ticket about failing tests, if there were no test to be ran ;)

Reply:
Yes sure, I do entirely realize that there are some tests outside, but clearly, some tests are located in the `pytest` directory:

```
$ tree pytest/
pytest/
├── Makefile
├── test_bytecode.py
├── test_disasm.py
├── test_instructions.py
├── test_load_file.py
├── test_opcode.py
├── test_std.py
├── testdata
│   ├── big_dict-3.6.right
│   ├── fstring-3.6.right
│   ├── multi-fn-2.7.pyc
│   ├── multi-fn-3.6.pyc
│   ├── multi-fn.py
│   ├── pypy_lambda.right
│   └── raise-3.0.right
└── todo-test.py

1 directory, 15 files
```

As far as I can see, there are tests in `test`, `pytest` and probably some in `test_unit`

Reply:
hmm, you're right. I'll change.